vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/experiments/experiment/ExperimentDetailsComponent.java

```diff
@@ -190,12 +192,27 @@ public Handler(ProjectInformationService projectInformationService,
       this.projectInformationService = projectInformationService;
       this.experimentInformationService = experimentInformationService;
       addCloseListenerForAddVariableDialog();
+
       experimentalGroupsLayoutBoard.setExperimentalGroupCommandListener(it -> {
         fillExperimentalGroupDialog();
-        experimentalGroupsDialog.open();
+        handleAddExperimentalGroups();
       });
     }
 
+    private void handleAddExperimentalGroups() {
+      List<ExperimentalVariable> variables = experimentInformationService.getVariablesOfExperiment(
+          experimentId);
+      if(!variables.isEmpty()) {
+        experimentalGroupsDialog.open();
+      } else {
+        experimentSheet.setSelectedIndex(0);
+        InformationMessage successMessage = new InformationMessage("No experimental variables are defined",
+            "Please define all of your experimental variables before adding groups.");
+        StyledNotification notification = new StyledNotification(successMessage);
+        notification.open();
+      }
+    }
```

Comment:
Should we not just show the user that no variables are yet defined and provide a link to do so?